src/game.cpp

```diff
@@ -3022,8 +3022,7 @@ bool game::handle_action()
                     }
                     as_m.text = data.str();
                 }
-                if( (u.has_item_with_flag("ALARMCLOCK") || u.has_bionic("bio_watch")) &&
-                    !(u.hunger < -60) ) {
+                if  ( u.has_alarm_clock() && !(u.hunger < -60) ) {
```

Comment:
Sorry about this oversight. I missed that check in this spot. The two checks here were previously inconsistent. The bionic check should probably have been included in both, and in my update.